quint/src/effects/inferrer.ts

```diff
@@ -91,24 +91,6 @@ export class EffectInferrer implements IRVisitor {
     }
 
     switch (def.kind) {
-      case 'param': {
```

Comment:
No need for this special treatment/workaround anymore now that lambdas have their own IDs.